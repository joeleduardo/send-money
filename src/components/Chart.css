.status {
  display: flex;
  justify-content: center;
  align-items: center;
}

.amounts {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.status span {
  color: #999999;
  font-size: .8rem;
}