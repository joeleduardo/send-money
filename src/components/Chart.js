import React from 'react';
import { Doughnut } from 'react-chartjs-2';
import './Chart.css';

class chartComponent extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      labels: props.chartLabels,
      data: props.chartData,
    }
  }

  componentDidUpdate(prevProps) {
    if(this.props.chartData !== prevProps.chartData) {
      this.setState({
        data: this.props.chartData
      })
    }
  }

  render() {
    const data = {
      labels: this.state.labels,
      datasets: [{
        data: this.state.data,
        backgroundColor: [
          '#ffb427',
          '#cccccc'
        ]
      }]
    }

    const options = {
      responsive: false,
      legend: {
        display: false
      }
    }

    const numberWithCommas = (amount) => {
      return amount.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
    }

    return (
      <div className='status'>
        <div className='amounts'>
          £ {numberWithCommas(this.state.data[0])}
          <span>{this.state.labels[0]}</span>
        </div>
        <Doughnut data={data} options={options} width={180}/>
        <div className='amounts'>
          £ {numberWithCommas(this.state.data[1])}
          <span>{this.state.labels[1]}</span>
        </div>
      </div>
    );
  }
}

export default chartComponent;