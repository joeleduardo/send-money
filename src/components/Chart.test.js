import React from 'react';
import { shallow } from 'enzyme';
import Chart from './Chart';

const chartLabels = ['Total sent', 'Left available']
const chartData = [0.00, 5000.00];

test('Chart component', () => {
  const cChart = shallow(<Chart chartLabels={chartLabels} chartData={chartData}/>);
  expect(cChart).toMatchSnapshot();
  cChart.setProps({ chartData: [100.00, 4900.00] });
  expect(cChart).toMatchSnapshot();
});
