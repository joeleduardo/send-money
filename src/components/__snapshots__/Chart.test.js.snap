// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <chartComponent
    chartData={
        Array [
            0,
            5000,
          ]
    }
    chartLabels={
        Array [
            "Total sent",
            "Left available",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="amounts"
>
          £ 
          0
          <span>
                    Total sent
          </span>
</div>,
        <Doughnut
          data={
                    Object {
                              "datasets": Array [
                                Object {
                                  "backgroundColor": Array [
                                    "#ffb427",
                                    "#cccccc",
                                  ],
                                  "data": Array [
                                    0,
                                    5000,
                                  ],
                                },
                              ],
                              "labels": Array [
                                "Total sent",
                                "Left available",
                              ],
                            }
          }
          options={
                    Object {
                              "legend": Object {
                                "display": false,
                              },
                              "responsive": false,
                            }
          }
          width={180}
/>,
        <div
          className="amounts"
>
          £ 
          5,000
          <span>
                    Left available
          </span>
</div>,
      ],
      "className": "status",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "£ ",
            "0",
            <span>
              Total sent
</span>,
          ],
          "className": "amounts",
        },
        "ref": null,
        "rendered": Array [
          "£ ",
          "0",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Total sent",
            },
            "ref": null,
            "rendered": "Total sent",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "datasets": Array [
              Object {
                "backgroundColor": Array [
                  "#ffb427",
                  "#cccccc",
                ],
                "data": Array [
                  0,
                  5000,
                ],
              },
            ],
            "labels": Array [
              "Total sent",
              "Left available",
            ],
          },
          "options": Object {
            "legend": Object {
              "display": false,
            },
            "responsive": false,
          },
          "width": 180,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "£ ",
            "5,000",
            <span>
              Left available
</span>,
          ],
          "className": "amounts",
        },
        "ref": null,
        "rendered": Array [
          "£ ",
          "5,000",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Left available",
            },
            "ref": null,
            "rendered": "Left available",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="amounts"
>
            £ 
            0
            <span>
                        Total sent
            </span>
</div>,
          <Doughnut
            data={
                        Object {
                                    "datasets": Array [
                                      Object {
                                        "backgroundColor": Array [
                                          "#ffb427",
                                          "#cccccc",
                                        ],
                                        "data": Array [
                                          0,
                                          5000,
                                        ],
                                      },
                                    ],
                                    "labels": Array [
                                      "Total sent",
                                      "Left available",
                                    ],
                                  }
            }
            options={
                        Object {
                                    "legend": Object {
                                      "display": false,
                                    },
                                    "responsive": false,
                                  }
            }
            width={180}
/>,
          <div
            className="amounts"
>
            £ 
            5,000
            <span>
                        Left available
            </span>
</div>,
        ],
        "className": "status",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "£ ",
              "0",
              <span>
                Total sent
</span>,
            ],
            "className": "amounts",
          },
          "ref": null,
          "rendered": Array [
            "£ ",
            "0",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Total sent",
              },
              "ref": null,
              "rendered": "Total sent",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "datasets": Array [
                Object {
                  "backgroundColor": Array [
                    "#ffb427",
                    "#cccccc",
                  ],
                  "data": Array [
                    0,
                    5000,
                  ],
                },
              ],
              "labels": Array [
                "Total sent",
                "Left available",
              ],
            },
            "options": Object {
              "legend": Object {
                "display": false,
              },
              "responsive": false,
            },
            "width": 180,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "£ ",
              "5,000",
              <span>
                Left available
</span>,
            ],
            "className": "amounts",
          },
          "ref": null,
          "rendered": Array [
            "£ ",
            "5,000",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Left available",
              },
              "ref": null,
              "rendered": "Left available",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Chart component 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <chartComponent
    chartData={
        Array [
            100,
            4900,
          ]
    }
    chartLabels={
        Array [
            "Total sent",
            "Left available",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="amounts"
>
          £ 
          100
          <span>
                    Total sent
          </span>
</div>,
        <Doughnut
          data={
                    Object {
                              "datasets": Array [
                                Object {
                                  "backgroundColor": Array [
                                    "#ffb427",
                                    "#cccccc",
                                  ],
                                  "data": Array [
                                    100,
                                    4900,
                                  ],
                                },
                              ],
                              "labels": Array [
                                "Total sent",
                                "Left available",
                              ],
                            }
          }
          options={
                    Object {
                              "legend": Object {
                                "display": false,
                              },
                              "responsive": false,
                            }
          }
          width={180}
/>,
        <div
          className="amounts"
>
          £ 
          4,900
          <span>
                    Left available
          </span>
</div>,
      ],
      "className": "status",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "£ ",
            "100",
            <span>
              Total sent
</span>,
          ],
          "className": "amounts",
        },
        "ref": null,
        "rendered": Array [
          "£ ",
          "100",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Total sent",
            },
            "ref": null,
            "rendered": "Total sent",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {
            "datasets": Array [
              Object {
                "backgroundColor": Array [
                  "#ffb427",
                  "#cccccc",
                ],
                "data": Array [
                  100,
                  4900,
                ],
              },
            ],
            "labels": Array [
              "Total sent",
              "Left available",
            ],
          },
          "options": Object {
            "legend": Object {
              "display": false,
            },
            "responsive": false,
          },
          "width": 180,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "£ ",
            "4,900",
            <span>
              Left available
</span>,
          ],
          "className": "amounts",
        },
        "ref": null,
        "rendered": Array [
          "£ ",
          "4,900",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Left available",
            },
            "ref": null,
            "rendered": "Left available",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="amounts"
>
            £ 
            100
            <span>
                        Total sent
            </span>
</div>,
          <Doughnut
            data={
                        Object {
                                    "datasets": Array [
                                      Object {
                                        "backgroundColor": Array [
                                          "#ffb427",
                                          "#cccccc",
                                        ],
                                        "data": Array [
                                          100,
                                          4900,
                                        ],
                                      },
                                    ],
                                    "labels": Array [
                                      "Total sent",
                                      "Left available",
                                    ],
                                  }
            }
            options={
                        Object {
                                    "legend": Object {
                                      "display": false,
                                    },
                                    "responsive": false,
                                  }
            }
            width={180}
/>,
          <div
            className="amounts"
>
            £ 
            4,900
            <span>
                        Left available
            </span>
</div>,
        ],
        "className": "status",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "£ ",
              "100",
              <span>
                Total sent
</span>,
            ],
            "className": "amounts",
          },
          "ref": null,
          "rendered": Array [
            "£ ",
            "100",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Total sent",
              },
              "ref": null,
              "rendered": "Total sent",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {
              "datasets": Array [
                Object {
                  "backgroundColor": Array [
                    "#ffb427",
                    "#cccccc",
                  ],
                  "data": Array [
                    100,
                    4900,
                  ],
                },
              ],
              "labels": Array [
                "Total sent",
                "Left available",
              ],
            },
            "options": Object {
              "legend": Object {
                "display": false,
              },
              "responsive": false,
            },
            "width": 180,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "£ ",
              "4,900",
              <span>
                Left available
</span>,
            ],
            "className": "amounts",
          },
          "ref": null,
          "rendered": Array [
            "£ ",
            "4,900",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Left available",
              },
              "ref": null,
              "rendered": "Left available",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
